.github/workflows/unit_tests.yml

```diff
@@ -286,7 +286,7 @@ jobs:
           while [ $i -ne 5 ]; do
             i=$(( $i+1 ))
             echo "Attempt $i/5 to run test targets"
-            bazel test --keep_going --remote_http_cache=$BAZEL_REMOTE_CACHE_URL --google_credentials=./config/oppia-dev-workflow-remote-cache-credentials.json -- $BAZEL_TEST_TARGETS
+            bazel test --keep_going --test_timeout=600 --remote_http_cache=$BAZEL_REMOTE_CACHE_URL --google_credentials=./config/oppia-dev-workflow-remote-cache-credentials.json -- $BAZEL_TEST_TARGETS
```

Comment:
**Try / TODO:**
If the broader instrumentation filter range is causing the issue, try narrowing it down more!

**eg:**

- for target - //utility/src/test/java/org/oppia/android/util/parser/math:MathModelTest -> //utility/src/test/java/org/oppia/android/util/parser/math/... (not sure if that's to be done with test or actual source file (if source) -> (same logic to map tests but in reverse) //utility/src/main/java/org/oppia/android/util/parser/math/...

- for target - //app:src/sharedTest/java/org/oppia/android/app/home/WelcomeViewModelTest -> //app/... (basically separating the package structure by ':')

Reply:
Should this be reverted? It doesn't seem relevant to this PR (unless I'm missing something).

Reply:
Sorry @BenHenning if this https://github.com/oppia/oppia-android/pull/5453#issuecomment-2199847753 was missed earlier or if I'm just sending a duplicate. Because of timeout issues, I had to introduce `--test_timeout=600`.

The provided comment was something I thought could be a workaround, but I don't think that could help with the tests for the reasons mentioned above so I opted to leave it unresolved to explore other potential possibilities.

Reply:
Inline with the discussion at the meeting -- the timeouts are more likely  to be an issue with the test setup than the CI configs, so  you will need to revert this and below. 

For instance, I pulled your changes and ran the RunCoverageTest on my Mac, and had similar timeouts. These yml configs have no impact on running tests locally, so adjusting the timeout on CI is more likely burying the real problem.

Reply:
Hey @adhiamboperes, thanks for testing it. 
So currently there are 2 issues _(provided in the Android CLaM meeting chat thread)_.

---

### Error with fetching in ci (not related to this current comment thread)
_(not related to `--test_timeout` issue)_

The issue we discussed in the meeting was related to the [ERROR: An error occurred during the fetch of repository 'remotejdk11_linux':](https://github.com/oppia/oppia-android/actions/runs/9860889216/job/27228421626?pr=5453#step:17:348) and that might be due to some mismatch of configurations with the `TestBazelWorkspace` setups (as you mentioned) which I am still trying to figure out. 

To give a bit of context, I tried running the **RunCoverageTest.kt** locally a couple of times and they do pass locally. So, its somewhere in the setup that has to have some mismatch configurations with ci, and causing the underlying issue.

![image](https://github.com/oppia/oppia-android/assets/122200035/959e2d75-1402-4a8e-96b0-72fde266b7df)

![image](https://github.com/oppia/oppia-android/assets/122200035/6632a56f-4478-46aa-85bb-47e05f2dd49a)


---

### --test_timeout
_(related to the current comment thread)_

Regarding the introduction of **--test_timeout**, I was experimenting with a few solutions, and this was one potential fix I found. However, it was **not intended to be merged** without further discussion with Ben. To keep this issue on our radar, I left the comment unresolved. I plan to discuss the underlying issues further to determine the best course of action.

Reply:
Oh, also missed mentioning this in the previous comment.

> For instance, I pulled your changes and ran the RunCoverageTest on my Mac, and had similar timeouts. These yml configs have no impact on running tests locally

Yeah! that's the case, so even with the local test it fails with the **SIGTERM** error. I had to add `--test_timeout` to the bazel test to give it the time to actually complete the test, and I replicated the same in the ci.

Reply:
This should now be addressed with replacing it with `size = "large"` for specific test targets as mentioned in https://github.com/oppia/oppia-android/pull/5453#issuecomment-2223125478

Reply:
Resolving this comment since the `--test_timeout` has been reverted and replaced by setting `size = "large"` for the specific test target.